WooCommerce/Classes/Extensions/UIImage+Dot.swift

```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+extension UIImage {
+    /// Draws a dot at the top right corner of a new image along with the original image.
+    ///
+    /// - Parameters:
+    ///   - imageOrigin: the origin where self image is located.
+    ///   - finalSize: the size of the returned image.
+    ///   - dotDiameter: the diameter of the dot.
+    ///   - dotColor: the color of the dot.
+    /// - Returns: an image with the original image at the given origin and a dot at the top right of the given diameter and color.
+    func imageWithTopRightDot(imageOrigin: CGPoint, finalSize: CGSize, dotDiameter: CGFloat, dotColor: UIColor) -> UIImage? {
```

Comment:
It seems likely that we are going to use this for now in the dashboard, would it make sense to expose default values for the dot's diameter and colour to fulfil whatever is needed in the dashboard? It might make the call site slightly less verbose (on the other hand, it'd make it less explicit as well, so 🤷‍♂ )

Reply:
yea the interface looks a bit verbose, but I think it'd be more clear making these parameters explicit because the placement of the dot relative to the original image depends heavily on the image. For example, the gift icon is able to have ~5px of overlap with the red dot because of its space at the top right corner. If the icon has a different shape, we might have to use a different coordinate for the image origin. For the diameter and color of the dot, I can move them to the default value

<img width="298" alt="Screen Shot 2019-09-02 at 10 33 50 PM" src="https://user-images.githubusercontent.com/1945542/64121595-d8c29300-cdd1-11e9-90c1-506440d7a4cd.png">